localstack/aws/forwarder.py

```diff
@@ -65,6 +67,36 @@ def HttpFallbackDispatcher(provider: object, forward_url_getter: Callable[[], st
     return ForwardingFallbackDispatcher(provider, get_request_forwarder_http(forward_url_getter))
 
 
+class ContextModifier(Protocol):
+    """
+    This protocol is used to apply a specific modification to a request context.
+    It does not modify the original context; it creates a newly one instead.
+    """
+
+    def modify_context(self, context: RequestContext) -> RequestContext:
+        ...
+
+
+class RegionContextModifier(ContextModifier):
```

Comment:
note on protocols vs abstract classes: extending a protocol class means creating a _new_ protocol that is based on the original one. in other words: by inheriting from `ContextModifier`, `RegionContextModifier` does not become an implementation of `ContextModifier`, it becomes a new protocol.